apps/ejabberd/src/ejabberd_admin.erl

```diff
@@ -44,7 +44,8 @@
          dump_to_textfile/1, dump_to_textfile/2,
          mnesia_change_nodename/4,
          restore/1, % Still used by some modules%%
-         get_loglevel/0]).
+         get_loglevel/0,
+         maybe_join_cluster/1, maybe_leave_cluster/0]).
```

Comment:
Should we export `maybe_*` functions as public API? In case of public interface I'd be for making the module API calls shorter. If after doing the checks (the "maybe"s hint at) the actual implementation needs to be put into deeper functions, we can call them `do_*` (`do_join_cluster`...).